packages/web-react/src/components/Dropdown/DropdownTrigger.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { ElementType } from 'react';
+import { useStyleProps } from '../../hooks';
+import { StyleProps } from '../../types';
+import { useDropdownContext } from './DropdownContext';
+import { useDropdownAriaProps } from './useDropdownAriaProps';
+import { useDropdownStyleProps } from './useDropdownStyleProps';
+
+interface DropdownTriggerProps extends StyleProps {
+  elementType?: ElementType | string;
+  children: string | React.ReactNode | ((props: { isOpen: boolean }) => React.ReactNode);
+}
+
+const DropdownTrigger = ({ elementType = 'button', children, ...rest }: DropdownTriggerProps) => {
+  const { id, isOpen, onToggle, fullWidthMode, triggerRef } = useDropdownContext();
+
+  const Component = elementType;
+
+  const { classProps } = useDropdownStyleProps({ isOpen, ...rest });
+  const { styleProps: triggerStyleProps } = useStyleProps({
+    UNSAFE_className: classProps.triggerClassName,
+  });
+  const { triggerProps } = useDropdownAriaProps({ id, isOpen, toggleHandler: onToggle, fullWidthMode });
+
+  return (
+    <Component id={id} {...rest} ref={triggerRef} {...triggerStyleProps} {...triggerProps}>
+      {typeof children === 'function' ? children({ isOpen }) : children}
+    </Component>
+  );
+};
+
+DropdownTrigger.defaultProps = {
```

Comment:
Do we need to set `defaultProps` when the default value is already set?

Reply:
Yes, ESLint requires this to be set. After a discussion in the office, we will leave it there.